### DISCUSSION TITLE
How works images storage ?

### DISCUSSION THREAD

**Discussion by rleffd** [artifacthub/hub#4130] [Q&A] [answered]
Hello,

I'm creating some automation about organizations dans I'm trying to set a logo on organizations.
Where are stored images ? File sytem ? DB ? Should I create a persistent volume to store all images (profiles and orgs images) ?

How can I send images from an automation ?

Thanks.

**Comment by tegioz** [comment]
In the database. You can find some more details about how they are managed in the [img package](https://github.com/artifacthub/hub/tree/master/internal/img).

You could automate it this way:

- Upload the image using the /api/v1/images endpoint. Post the image blob to it and it'll return its id.
- Update the organization profile with the image id you obtained (using the API again).

We use a form of content-addressable storage for images, which allows us to cache them aggressively (to avoid hitting the database).

**Reply by rleffd** [reply]
I can't find any docs about /api/v1/images in openapi. Is it a simple POST to /api/v1/images with api key/secrets in headers? 

```bash
curl -v -X POST -H "Content-Type=image/png" -H "X-API-KEY-ID=${X-API-KEY-ID}" -H "X-API-KEY-SECRET=${X-API-KEY-SECRET}" --upload-file ./logo.png  https://xxxxxx/api/v1/images

Forbidden - CSRF token not found in request
```

**Reply by cynthia-sg** [reply, marked as answer]
You can try this:

```
curl -X 'POST' \
  'https://xxxx/api/v1/images' \
  -H 'accept: */*' \
  -H 'X-API-KEY-ID: xxxx' \
  -H 'X-API-KEY-SECRET: xxxx' \
  -H 'Content-Type: image/png' \
  --data-binary "@/path/to/image.png"
```

**Reply by tegioz** [reply]
That endpoint is not documented because it's not fully stable and it's subject to change @rleffd.

**Reply by rleffd** [reply]
thanks. as it is not documented, is there a DELETE method for this image ? Or a simple delete sql command suffice to remove an image that is not used by any other organization ?

**Reply by tegioz** [reply]
You can delete them directly from the database (`image` and `image_version` tables).
